controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1516,10 +1513,160 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
+    case 117:
+        return std::vector<uint8_t>{40, 48};
+    case 118:
+        return std::vector<uint8_t>{52, 56, 60, 64};
+    case 119:
+        return std::vector<uint8_t>{52, 60};
+    case 120:
+        return std::vector<uint8_t>{56, 64};
+    case 121:
+        return std::vector<uint8_t>{100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144};
+    case 122:
+        return std::vector<uint8_t>{100, 108, 116, 124, 132, 140};
+    case 123:
+        return std::vector<uint8_t>{104, 112, 120, 128, 136, 144};
+    case 124:
+        return std::vector<uint8_t>{149, 153, 157, 161};
+    case 125:
+        return std::vector<uint8_t>{149, 153, 157, 161, 165, 169};
+    case 126:
+        return std::vector<uint8_t>{149, 157};
+    case 127:
+        return std::vector<uint8_t>{153, 161};
+    case 180:
+        return std::vector<uint8_t>{1, 2, 3, 4, 5, 6};
+    default:
+        LOG(ERROR) << "reserved operating class " << operating_class;
+        return std::vector<uint8_t>();
+    }
+}
+
+/**
+ * @brief parse radio capabilities into supported channels which can be saved in the database
+ *        Currently this function only prints the supported operating classes and the non operable
+ *        channel list per each operating class.
+ *        TODO - as part of channel selection task, convert operating classes to supported channels
+ *        so they can be stored in the DB using database.set_hostap_supported_channels()
+ * 
+ * @param radio_caps 
+ * @param supported_channels 
+ * @return true 
+ * @return false 
+ */
+std::vector<beerocks_message::sWifiChannel> master_thread::autoconfig_wsc_parse_radio_caps(
+    std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_caps)
+{
+    // read all operating class list
+    std::vector<beerocks_message::sWifiChannel> supported_channels;
+    auto operating_classes_list_length = radio_caps->operating_classes_info_list_length();
+    if (operating_classes_list_length > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
+        LOG(WARNING) << "operating class info list larger then maximum supported channels";
+        operating_classes_list_length = beerocks::message::SUPPORTED_CHANNELS_LENGTH;
+    }
+    for (int oc_idx = 0; oc_idx < operating_classes_list_length; oc_idx++) {
+        std::stringstream ss;
+        auto operating_class_tuple = radio_caps->operating_classes_info_list(oc_idx);
+        if (!std::get<0>(operating_class_tuple)) {
+            LOG(ERROR) << "getting operating class entry has failed!";
+            return supported_channels;
+        }
+        auto &op_class_channels_rx = std::get<1>(operating_class_tuple);
+        auto operating_class       = op_class_channels_rx.operating_class();
+        auto channel_set           = operating_class_to_channel_set(operating_class);
+        for (auto c : channel_set) {
+            beerocks_message::sWifiChannel ch;
+            ch.channel = c;
+            supported_channels.push_back(ch);
```

Comment:
This part of this flow should be implemented as part of the channel selection full flow.
Besides, the fact that we have used channels list until now, does not say we have to continue working that way.
AFAIU, the Multi-AP messages use operating class instead of a channel, so maybe it will be more convenient to save it as it, instead of converting it back and forth.
IMO, print only what you get from the CMDU, and don't store it in the database. When we will implement channel selection we will know better.

Reply:
The only reason I did that is for the Conn map to show if the radio is 5ghz or 2.4ghz. I agree that we should store this as is without converting but we will do this refactoring as part of channel selection.
In the meanwhile it won't hurt to have this conversion even if it is only for the network map, since it's a nice thing to show Ina demo.
In fact, it will also allow us to use our steering algorithm before implementing full channel selection which is also nice to have and we plan to try to get this working for the demo.